src/rat/run_rat.py

```diff
@@ -143,41 +181,89 @@ def run_rat(config_fn, operational_latency=None):
-                    basins_metadata['BASIN','vic_init_state'].where(basins_metadata['BASIN','basin_name']!= basin, basins_metadata['BASIN','start'], inplace=True)
+                        basins_metadata['BASIN','vic_init_state'] = None
+                    # Find vic_init_state
+                    vic_init_state_date = get_vic_init_state_date(previous_end_date, imerg_latency, data_dir,region_name, basin_name)
+                    # Check if vic init state is not none, else raise error
+                    if not(config['BASIN']['vic_init_state']):
+                        raise Exception('No vic init state file was found from last run.')
+                    # Replace vic_init_state in basins metadata 
+                    basins_metadata['BASIN','vic_init_state'].where(basins_metadata['BASIN','basin_name']!= basin, vic_init_state_date, inplace=True)
+
+                     ## Adding [Basin][start] to metadata.columns if not there with with None value 
+                    if ('BASIN','start') not in basins_metadata.columns:
+                        basins_metadata['BASIN','start'] = None
+                    # Replace start date same as vic_init_date
+                    basins_metadata['BASIN','start'].where(basins_metadata['BASIN','basin_name']!= basin, vic_init_state_date, inplace=True)
+                    
+                    ## Adding [Basin][end] to metadata.columns if not there with with None value 
+                    if ('BASIN','end') not in basins_metadata.columns:
+                        basins_metadata['BASIN','end'] = None
+                    # Replace end date as today - operational latency
+                    operational_end_date = datetime.datetime.now().date() - datetime.timedelta(days=int(operational_latency))
+                    basins_metadata['BASIN','end'].where(basins_metadata['BASIN','basin_name']!= basin, operational_end_date, inplace=True)
+
                     # Updating rout_init_state to None for all.
-                    ## Adding [Basin][vic_init_state] to metadata.columns if not there with None value
+                    ## Adding [Basin][rout_init_state] to metadata.columns if not there with None value
                     if ('BASIN','rout_init_state') not in basins_metadata.columns:
                         basins_metadata['BASIN','rout_init_state'] = None
                     basins_metadata['BASIN','rout_init_state'] = None
+
                 except:
                     log.exception('Failed to update start and end date for RAT to run operationally. Please make sure RAT has been run atleast once before.')
                     return None
+            
+            else:
+                ## Calculation of gfs_days
+                start_date_value = basins_metadata['BASIN','start'].loc[basins_metadata['BASIN','basin_name']== basin]
+                end_date_value = basins_metadata['BASIN','end'].loc[basins_metadata['BASIN','basin_name']== basin]
```

Comment:
check for start and end date in config file if there

Reply:
Added the check and the scenario if the parameter is in config file and not in metadata.